### PR TITLE
created models and controllers, made basic test

### PR DESCRIPTION
**Pull request by a-nurmatov**

### CODE REVIEW

File: __test__/user.test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import supertest from "supertest";
+import mongoose from "mongoose";
+import { config } from "dotenv";
+
+config();
+
+describe("users", () => {
+  beforeAll(async () => {
+    await mongoose.disconnect();
+    await mongoose.connect(process.env.MONGO_URI);
```
Comment (AudriusBaltusis) [inline __test__/user.test.js]: Do not use production DB for unit tests. Please create some test DB for it

File: __test__/user.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    await mongoose.disconnect();
+  });
+
+  it("it returns 200 status code", async () => {
+    const res = await supertest(app).get("/users");
+    expect(res.statusCode).toEqual(200);
+  });
+  describe("register", () => {
+    test("/signup", async () => {
+      const res = await supertest(app).post("/users/signup").send({
```
Comment (AudriusBaltusis) [inline __test__/user.test.js:24]: If you are adding some records to DB, after tests you should wipe test data

File: api/models/transaction.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    amount: {
+      type: Number,
+      min: 1,
+      required: [true, "Amount is required"],
+    },
+    date: {
+      type: Date,
+      required: [true, "Date is required"],
+    },
+    file: {
```
Comment (AudriusBaltusis) [inline api/models/transaction.js]: `fileId`

File: api/models/transaction.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      enum: {
+        values: ["income", "expense"],
+        message: "{VALUE} is not valid",
+      },
+    },
+    name: {
+      type: String,
+      required: [true, "Name required"],
+    },
+    category: {
```
Comment (AudriusBaltusis) [inline api/models/transaction.js]: `categoryId`

**Review by AudriusBaltusis — CHANGES_REQUESTED** [review]

File: __test__/user.test.js (hunk 1 above)
Comment (a-nurmatov) [reply __test__/user.test.js]: ok I will change it, just a second

File: __test__/user.test.js (hunk 2 above)
Comment (a-nurmatov) [reply __test__/user.test.js:24]: yes it is my bad, I will correct it now

**Comment by a-nurmatov** [comment]
i fixed issues could you check it now?, thank you !

**Review by AudriusBaltusis — APPROVED** [review]

**Pull request merged**